src/realm/sync/client.cpp

```diff
@@ -1797,20 +1829,28 @@ void SessionWrapper::report_progress(bool only_if_new_uploadable_data)
     // uploadable_bytes is uploaded + remaining to upload, while downloadable_bytes
     // is only the remaining to download. This is confusing, so make them use
     // the same units.
-    std::uint_fast64_t total_bytes = downloaded_bytes + downloadable_bytes;
+    downloadable_bytes += downloaded_bytes;
 
     m_sess->logger.debug("Progress handler called, downloaded = %1, "
-                         "downloadable(total) = %2, uploaded = %3, "
-                         "uploadable = %4, reliable_download_progress = %5, "
-                         "snapshot version = %6",
-                         downloaded_bytes, total_bytes, uploaded_bytes, uploadable_bytes,
-                         m_reliable_download_progress, snapshot_version);
-
-    // FIXME: Why is this boolean status communicated to the application as
-    // a 64-bit integer? Also, the name `progress_version` is confusing.
-    std::uint_fast64_t progress_version = (m_reliable_download_progress ? 1 : 0);
-    m_progress_handler(downloaded_bytes, total_bytes, uploaded_bytes, uploadable_bytes, progress_version,
-                       snapshot_version);
+                         "downloadable = %2, uploaded = %3, "
+                         "uploadable = %4, snapshot version = %5",
+                         downloaded_bytes, downloadable_bytes, uploaded_bytes, uploadable_bytes, snapshot_version);
+
+    // FIXME take into account first initial reported value in this batch
+    double upload_estimate = uploadable_bytes > 0 ? (uploaded_bytes / double(uploadable_bytes)) : 1;
```

Comment:
nit, this is not really an estimate - we know for sure this is how many uploadable bytes we have and how many bytes we have yet to upload.

Reply:
yeah.. i've also mentioned this multiple time during scoping, but it's not really important. Since bytes will go away at some point with pbs, then the only thing which is deemed important is the progress value itself. And it's somewhat obscure number, but it's the only useful thing for user facing progress. And it can't be precise, especially we use the same api in object-store for flx download direction.

Reply:
so, is it fine to resolve or would you like to weight in on changing it somehow from the original scope?